python/mp/effects/sine_wave.py

```diff
@@ -4,7 +4,14 @@
 from mp import color
 
 class SineWave(effect.Effect):
-
+    """Sets the intensity of a set of beads by mapping a unit circle to
+    the beads in the bead_list then computing a sine wave. Each step
+    advances the angle by 2 * pi / number of beads.
+
+    knobs:
+    * period: the number of sine wave periods in the unit circle (can be float)
+    * direction: the distance (positive or negative float) to advance around the unit circle in one mainloop step
```

Comment:
Hmm, I think I see what you're doing here. As you define "knobs" this should inform the OSC/REST API. Goooootcha...